### ISSUE TITLE
0.0.3 release

### ISSUE THREAD

**Issue by cdrage** [streamlink/streamlink#116]
Been 21 days since the last release. Down for another one?

Now that we have rtmp updated + windows docs + debian deb package installer thingy, it makes sense.

Let me know if there any blockers on this, otherwise, I can do a release later tomorrow.

**Comment by gravyboat** [comment]
We aren't building the Windows installer automatically yet is my only concern for creating more manual work. @sbstp opinions?

**Comment by RangerCoder99** [comment]
Thank you so much for updating this project, and all your hard work on this!

**Comment by cdrage** [comment]
@gravyboat I should add this to the release / build script we have.

**Comment by sbstp** [comment]
I'd like to add some polish if you guys don't mind. Shouldn't be too long, it's just the config file thingy on Windows.

**Comment by gravyboat** [comment]
I'm fine with waiting @sbstp! Having the config file taken care of would be good for typical users (I am assuming that is what you are referring to, if not let me know).

**Comment by karlo2105** [comment]
Could you do something like chrippa did with automatically nightly build : http://livestreamer-builds.s3.amazonaws.com/livestreamer-latest-win32.zip

**Comment by sbstp** [comment]
Yeah, the first time you run streamlink, it would create the file in %APPDATA%, and I might also add a switch such as `streamlink --edit-config` that starts an editor (windows notepad, unix nano/vim/emacs/$EDITOR) with the config file. You're right though, it can wait.

**Comment by gravyboat** [comment]
@sbstp The editor idea is a good idea! Thanks for prioritizing the configuration file, it's super helpful to have that available. @karlo2105 personally I am not a fan of nightly builds, people use them and then ask for support when instead they could just compile it themselves if they want the latest release since everything is available.

**Comment by thinkpad4** [comment]
Sbstp, are you gonna make another windows .exe installer for 0.0.3 like you did for 0.0.2?

**Comment by sbstp** [comment]
Yes I will.

**Comment by thinkpad4** [comment]
<3

**Comment by karlo2105** [comment]
@gravyboat
Nightlybuild is interesting when you want to update streamlink locally altogether on Windows. 
Otherwise you are obliged to update step by step every part of streamlink. 
I like to have choice, nobody has problems with nightly builds with Livestreamer and no one is obliged to use nightlybuilds.

**Comment by dehesselle** [comment]
Hey guys,
I wanted to drop by and thank you for stepping in where Livestreamer left off. Also I'd like to mention that Streamlink is part of the [macOS app of Chatty](https://github.com/dehesselle/chatty) that I'm starting to maintain. I'm hoping that's fine with you?

**Comment by gravyboat** [comment]
@dehesselle Sounds good to me. I don't have any concerns about including it.

**Comment by cdrage** [comment]
@dehesselle that's awesome by us, if you need any addition macOS bugs / fixes feel free to open up any new issues and we'd be on it.

**Comment by cyluss** [comment]
Any progress?

**Comment by scottbernstein** [comment]
I think that all of this work @beardypig has been doing on the ustream plugin will be nice to have in the 0.0.3 release.

________________________________

**Comment by gravyboat** [comment]
@cyluss We're waiting on a few things to be finished prior to doing this, ustream would be nice to have, but more important to me is the Windows config file being available by default.

**Comment by cyluss** [comment]
So #81 is blocker, right?

**Comment by gravyboat** [comment]
@cyluss Correct.

**Comment by cdrage** [comment]
Hey @beardypig so I think it's time for another release (with the Windows .exe).

Do you think that the .exe is ready?

To be honest, I don't have access to a Windows machine at the moment for testing. But if you say it's working fine, I'm okay with the release.

I'll test Linux, you test Windows :)

Give me the go-ahead for the current code in master and I'll compile the .exe and do a release. If there's an issue (luckily) it's as simple creating a patch and doing another release.

**Comment by beardypig** [comment]
I actually use macOS most of the time, but I do use a Windows 10 VM for testing. The current master version was working well when I tested it, except you have to update the streamlinkrc files to use rtmpdump - hence my patch for the installer :-) 

Would be good if a couple of other users could test out the nightly...

**Comment by bastimeyer** [comment]
I just tested streamlink-latest.exe in my Windows 7 and 10 VMs.
What about #172 (streamlink.rc)? Shouldn't this be merged first before publishing a new release?

What I noticed about the installer:
- It creates an unnecessary start menu shortcut which might annoy some users
- Publisher infos are missing (name and website)
- The copy-file section should not be closed automatically (I'd like to review the file list). Looks like `pynsist` is using the [nsis 2.0 modern user interface](http://nsis.sourceforge.net/Docs/Modern%20UI%202/Readme.html). The `MUI_FINISHPAGE_NOAUTOCLOSE` constant needs to be defined for this. ([default nsis config](https://github.com/takluyver/pynsist/blob/master/nsist/pyapp.nsi))
- A final section could be added similar to the Livestreamer installer that suggests editing the config file and/or opening the publisher's website.

**Comment by beardypig** [comment]
@bastimeyer sounds good, do you want to make some/all of those changes?

**Comment by bastimeyer** [comment]
@beardypig 
Unfortunately, I'm not too familiar with nsis. How can I change the pynsist default config (or extend it / remove stuff)?

Regarding the shortcut (pynsist docs):

> One shortcut will always be generated for the application

😕

**Comment by bastimeyer** [comment]
Ok, nvm, figured it out.

The shortcut can be removed by overriding the whole [`install_shortcuts`](https://github.com/takluyver/pynsist/blob/1.9/nsist/pyapp.nsi#L70-L84) block with an empty one.

The constants can be added by **prepending** them to the [`modernui`](https://github.com/takluyver/pynsist/blob/1.9/nsist/pyapp.nsi#L14-L28) block.
```
!define MUI_FINISHPAGE_NOAUTOCLOSE
!define MUI_UNFINISHPAGE_NOAUTOCLOSE
```

I can't manage to add the publisher data, though, as it would require rewriting the whole `sections` block which contains [these lines](https://github.com/takluyver/pynsist/blob/1.9/nsist/pyapp.nsi#L98-L105). (there's no nested block, so there's no *nice* way (without forking pynsist))

Publisher data:
```
WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "DisplayVersion" "..."
WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "Publisher" "..."
WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "URLInfoAbout" "..."
WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "HelpLink" "..."
```

I can add a PR once #172 gets merged, so that no conflicts occur. It's past 1am here, if you want to do it yourself now, just go for it. 😄

**Comment by gravyboat** [comment]
I merged #172.

**Comment by beardypig** [comment]
@bastimeyer seems pretty straight forward then. It might be possible to tackle the publisher info on the end of the install_commands block... Not ideal, but could be tidied up later if pynsist gets a block around the add/remove stuff, or adds a publisher info section to the pynsist configuration...

**Comment by gravyboat** [comment]
At this point I think we've addressed most of the concerns people had about the 0.0.3 release and it's been over a month since the last release. If there are no further concerns and everyone is happy with what we've done let's get this built. We can always release 0.0.4 much sooner if something crops up that is problematic.

**Comment by sbstp** [comment]
@gravyboat do you have push access to pypi? If not we need @cdrage for the 0.0.3 release!

**Comment by gravyboat** [comment]
@sbstp We need @cdrage to do it.

**Comment by sbstp** [comment]
We need more people to have access to pypi so that the livestreamer story doesn't repeat itself.

**Comment by gravyboat** [comment]
Agreed. I know sometimes people don't hop on GitHub during the weekend (understandably) so it might need to wait till Monday.

**Comment by cdrage** [comment]
@sbstp @gravyboat I'll give you guys access to pypi once i'm at a computer on monday. send me your email details @ charlie (at) charliedrage (dot) com if they're different than what's on github.

**Comment by cdrage** [comment]
0.1.0 released!

**Comment by gravyboat** [comment]
I'm just following up from this note 22 days ago from @cdrage regarding pypi access. A quick google search doesn't seem to elude to the fact that multiple people can share a repo, is this even possible?

**Comment by beardypig** [comment]
It is possible, there is a "roles" section under the "edit" section for the package on PyPi. You can add another PyPi user as a maintainer. I guess the URL is https://pypi.python.org/pypi?:action=role_form&package_name=streamlink :-)

**Comment by gravyboat** [comment]
Good find @beardypig, not sure why I couldn't find anything. We should probably add at least 2 more people to the pypi repo for bus factor and all that.

**Comment by cdrage** [comment]
@gravyboat @beardypig what are your pypi usernames?

**Comment by beardypig** [comment]
You guessed it, mine is beardypig :)

**Comment by cdrage** [comment]
@beardypig added you :)

**Comment by beardypig** [comment]
Confirmed. Thanks :)

**Comment by gravyboat** [comment]
@cdrage The same as my github username, gravyboat.
